modules/backendscheduler/provider/compaction.go

```diff
@@ -263,10 +192,6 @@ func (p *CompactionProvider) getNextBlockIDs(_ context.Context) ([]string, bool)
 	}
 
 	for _, b := range toBeCompacted {
-		if p.sched.HasBlocks([]string{b.BlockID.String()}) {
```

Comment:
Skipping blocks which are in flight is covered by the `newBlockSelector`. No need to duplicate it here.